enhance/hurlfmt: remove depreciated --format option

bp7968h opened the pull request:
This PR removes the `--format` cli args for `hurlfmt`

Closes #3445

jcamiel commented on .gitignore:
```diff
@@ -3,6 +3,7 @@
 *.profraw
 .mypy_cache/
+*.log
 
```
This file should not be changed.

jcamiel commented:
Hi thanks for the change @bp7968h could you squash your commits so we have only one commit for this PR ?  Thanks a lot 🙏

jcamiel commented:
@bp7968h You will also need to sign your commit, as you can see in the [CONTRIBUTING guide](https://github.com/Orange-OpenSource/hurl/blob/master/CONTRIBUTING.md). Your commits should have a "verified" status:

![commit](https://github.com/user-attachments/assets/07f122e7-c3a1-4bc6-b195-b51783ede9fc)

bp7968h commented:
Hi @jcamiel, I have squashed and signed the commits as well, can you please verify. Thank you very much for your guidance.

bp7968h commented:
@jcamiel , the ci failed again, can you please guide me here, what does the below mean which is displayed in the CI:

```bash
index dd18217..07a7ba1 100644
--- a/packages/hurlfmt/src/cli/options/commands.rs
+++ b/packages/hurlfmt/src/cli/options/commands.rs
@@ -30,6 +30,7 @@ pub fn check() -> clap::Arg {
     clap::Arg::new("check")
         .long("check")
         .help("Run in check mode")
+        .conflicts_with("format")
         .conflicts_with("output")
         .action(clap::ArgAction::SetTrue)
 }
You must regenerate files by running 'bin/spec/options/generate_all.py'
and commit them
```

jcamiel commented:
Hi @bp7968h 

Some code are generated (for instance, the completion shell scripts). The CI/CD checks that the code have been regenerated, so you just have to run this script:

```
$ python3 bin/spec/options/generate_all.py
```

bp7968h commented:
> Hi @bp7968h
> 
> Some code are generated (for instance, the completion shell scripts). The CI/CD checks that the code have been regenerated, so you just have to run this script:
> 
> ```
> $ python3 bin/spec/options/generate_all.py
> ```
> 
> And amend your commit with the updated files.

Hi @jcamiel, thank you for your direction, just wanted to confirm as I want to nail down these PR during the weekend;

- I edit the code and check if it's all right with `cargo check/cargo build`.
- Now, to check if it aligns with the CI/CD, I run the below code in the `venv`
```bash
cd integration/hurl
python3 server.py > server.log 2>&1 &
python3 tests_ssl/ssl_server.py 8001 tests_ssl/certs/server/cert.selfsigned.pem false > server-ssl-selfsigned.log 2>&1 &
python3 tests_ssl/ssl_server.py 8002 tests_ssl/certs/server/cert.pem false > server-ssl-signedbyca.log 2>&1 &
python3 tests_ssl/ssl_server.py 8003 tests_ssl/certs/server/cert.selfsigned.pem true > server-ssl-client-authent.log 2>&1 &
python3 tests_unix_socket/unix_socket_server.py > server-unix-socket.log 2>&1 &
squid_conf="cache deny all\ncache_log /dev/null\naccess_log /dev/null\nhttp_access allow all\nhttp_port 127.0.0.1:3128\nrequest_header_add From-Proxy Hello\nreply_header_add From-Proxy Hello"
(echo -e "${squid_conf}" | sudo squid -d 2 -N -f /dev/stdin | sudo tee proxy.log 2>&1) &
jobs
```
- After the abose certain ports are opened, and I run the integration test with `cd integration/hurl && python3 integration.py` right.

These are the steps required right, can you please mention if I am missing something. Thank you very much.

And sorry for the failed CI/CD.

jcamiel commented:
Hi @bp7968h 

There are the full steps that you _could_ do to be sure that the integration tests. In this PR, integration tests are all green, it's just the check on generated source files that is missing (you can see my previous comment). You can try to run integration tests locally of course, but that's not a must!

bp7968h commented:
> Hi @bp7968h
> 
> There are the full steps that you _could_ do to be sure that the integration tests. In this PR, integration tests are all green, it's just the check on generated source files that is missing (you can see my previous comment). You can try to run integration tests locally of course, but that's not a must!

bp7968h commented:
Hi @jcamiel, so meaning I just generate the files `python3 bin/spec/options/generate_all.py` and do I like add those files in the commit itself?
```bash
bpandit@fedora:~/oss/hurl$ git status
On branch enhance/hurlfmt-3445
Your branch is up to date with 'origin/enhance/hurlfmt-3445'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   completions/_hurlfmt
	modified:   completions/_hurlfmt.ps1
	modified:   completions/hurlfmt.bash
	modified:   completions/hurlfmt.fish
```

jcamiel commented:
Exactly, as we have remove an option of the `hurlfmt` cli, the completions scripts have to be updated to reflect this!

bp7968h commented:
> Exactly, as we have removed an option of the `hurlfmt` cli, the completions scripts have to be updated to reflect this!

Thank you veryyyy much for your help, really appreciate it.

bp7968h commented:
Sorry @jcamiel, seems the CI failed again, but I did add all the generated files and the reason seems the same.

jcamiel commented:
Don't worry, the CI/CD can take a little time until completion, we can wait until completion and we'll see then, I'm confident 😎

jcamiel commented:
/accept

jcamiel commented:
/accept
